ui/src/app/tasks/task-executions-destroy/task-executions-destroy.component.spec.ts

```diff
@@ -0,0 +1,242 @@
+import { RolesDirective } from '../../auth/directives/roles.directive';
+import { TasksService } from '../tasks.service';
+import { MockTasksService } from '../../tests/mocks/tasks';
+import { NotificationService } from '../../shared/services/notification.service';
+import { LoggerService } from '../../shared/services/logger.service';
+import { BlockerService } from '../../shared/components/blocker/blocker.service';
+import { TaskExecution } from '../model/task-execution';
+import { TaskExecutionsDestroyComponent } from './task-executions-destroy.component';
+
+/**
+ * Test {@link TaskExecutionsDestroyComponent}.
+ *
+ * @author Damien Vitrac
+ */
+describe('TaskExecutionsDestroyComponent', () => {
+
+  let component: TaskExecutionsDestroyComponent;
+  let fixture: ComponentFixture<TaskExecutionsDestroyComponent>;
+  const notificationService = new MockNotificationService();
+  const authService = new MockAuthService();
+  const bsModalRef = new BsModalRef();
+  const tasksService = new MockTasksService();
+  const loggerService = new LoggerService();
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      declarations: [
+        TaskExecutionsDestroyComponent,
+        StreamDslComponent,
+        TruncatePipe,
+        RolesDirective
+      ],
+      imports: [
+        FormsModule,
+        ModalModule.forRoot(),
+        TooltipModule.forRoot(),
+        ReactiveFormsModule,
+        RouterTestingModule.withRoutes([])
+      ],
+      providers: [
+        { provide: AuthService, useValue: authService },
+        { provide: BsModalRef, useValue: bsModalRef },
+        { provide: NotificationService, useValue: notificationService },
+        { provide: TasksService, useValue: tasksService },
+        { provide: LoggerService, useValue: loggerService },
+        BlockerService
+      ]
+    }).compileComponents();
+  }));
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(TaskExecutionsDestroyComponent);
+    component = fixture.componentInstance;
+    notificationService.clearAll();
+  });
+
+  describe('1 tasks destroy', () => {
+    const mock = [
+      TaskExecution.fromJSON({
+        executionId: '1',
+        exitCode: 0,
+        taskName: 'foo1',
+        startTime: null,
+        endTime: null,
+        exitMessage: null,
+        arguments: [],
+        jobExecutionIds: [],
+        errorMessage: null,
+        externalExecutionId: null
+      })
+    ];
+
+    beforeEach(() => {
+      component.open({ taskExecutions: mock });
+      fixture.detectChanges();
+    });
+
+    it('Should display a dedicate message', () => {
+      const message: HTMLElement = fixture.debugElement.query(By.css('.modal-body')).nativeElement;
+      expect(message.textContent).toContain(`This action will destroy and delete the task execution`);
+      expect(message.textContent).toContain(`1`);
+    });
+
+    it('Should call the service on validate destroy', () => {
```

Comment:
Wording: `Should call the service after validating the button click to destroy the task execution`